<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Correlation and Consensus-Based Cross-Taxonomy Network Analysis (C3NA) • C3NA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Correlation and Consensus-Based Cross-Taxonomy Network Analysis (C3NA)">
<meta property="og:description" content="The correlation and consensus-based cross-taxonomy network analysis 
  (C3NA) is an R package for analyzing compositional microbial data, particularly 
  the 16S rRNA amplicon sequencing data. The C3NA package contains tools to facilitate 
  analysis and shiny interactive applications to uncover complex taxa-taxa 
  associations within one or a comparison between two phenotypes. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">C3NA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/C3NA.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/rMD1_Demo.html">Tutorial with Demo</a>
    </li>
    <li>
      <a href="articles/rMD2_DA.html">Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="articles/rMD3_FAQ.html">FAQ</a>
    </li>
    <li>
      <a href="articles/rMD4_ModEvals.html">Tutorial on Using Module Evaluation Shiny</a>
    </li>
    <li>
      <a href="articles/rMD5_ComparePhenotypes.html">Tutorials on Using Compare Phenotypes Shiny</a>
    </li>
    <li>
      <a href="articles/rMD6_C3NAObjectDescription.html">C3NA Objects and Description</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zhouLabNCSU/C3NA/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="correlation-and-consensus-based-cross-taxonomy-network-analysis-c3na-">Correlation and Consensus-based Cross-taxonomy Network Analysis (C3NA) <a href="https://github.com/zhouLabNCSU/C3NA/" class="external-link"><img src="./reference/figures/RPackageLogo.png" align="right" width="120" height="100%"></a>
<a class="anchor" aria-label="anchor" href="#correlation-and-consensus-based-cross-taxonomy-network-analysis-c3na-"></a>
</h1></div>
<!-- badges: start -->

<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ol style="list-style-type: decimal">
<li><a href="#Description">Description</a></li>
<li><a href="#installation">Installation</a></li>
<li><a href="#workflow">Workflow</a></li>
<li><a href="#package-documents-and-tutorials">Package Documents and Tutorials</a></li>
<li><a href="#shiny-domonstration">Shiny Demonstration</a></li>
</ol>
</div>
<div class="section level2">
<h2 id="description">Description<a class="anchor" aria-label="anchor" href="#description"></a>
</h2>
<p>C3NA is a open-source R package for co-occurence patterns investigation for compositional microbial sequencing data. C3NA used a consensus-based approach to cluster taxa from multiple taxonomic levels into modules and it can be used to conduct differential abundance network analysis between two phenotypes. <br> Please check out the <a href="https://zhoulabncsu.github.io/C3NA/articles/C3NA.html" class="external-link">Get started</a> to see the website for the C3NA packge.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<pre><code><span class="co"># install.packages("devtools")</span>
<span class="co"># devtools::install_github("zhouLabNCSU/C3NA")</span>
<span class="co"># library(C3NA)</span></code></pre>
<p>Please note due to many dependencies required for the C3NA, particularly the shiny elements to function properly, please visit the <a href="https://zhoulabncsu.github.io/C3NA/articles/C3NA.html" class="external-link">Get started</a> to install the dependencies before install the C3NA package. <em>C3NA has been tested using the GitHub workflow using the up-to-date macOS, windows and Ubuntu using R version 4.1.</em></p>
</div>
<div class="section level2">
<h2 id="workflow">Workflow<a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<p><img src="reference/figures/C3NAPipeline.png" align="center"></p>
</div>
<div class="section level2">
<h2 id="package-documents-and-tutorials">Package Documents and Tutorials<a class="anchor" aria-label="anchor" href="#package-documents-and-tutorials"></a>
</h2>
<p>The package homepage contains tutorials and detailed guide for the C3NA pipeline. - <a href="https://zhoulabncsu.github.io/C3NA/index.html" class="external-link">Package Home Page</a></p>
</div>
<div class="section level2">
<h2 id="shiny-demonstration">Shiny Demonstration<a class="anchor" aria-label="anchor" href="#shiny-demonstration"></a>
</h2>
<p>There are two interaction shiny applications: Module Evaluations and Compare Two Phenotypes</p>
<div class="section level3">
<h3 id="module-evaluations-shiny">Module Evaluations Shiny<a class="anchor" aria-label="anchor" href="#module-evaluations-shiny"></a>
</h3>
<p><img src="reference/figures/Shiny1PanelGuide.jpg" align="center"><br> The evaluation of the modules should follow the sequence: 2 &gt;&gt; 3 &gt;&gt; 4</p>
<ul>
<li>
<img src="https://via.placeholder.com/15/4d4d4d/000000?text=+" alt="#4d4d4d"> 1: Shiny panel navigation, including the ‘Consensus-Based Evaluations’ and ‘Download Panel’.</li>
<li>
<img src="https://via.placeholder.com/15/c13004/000000?text=+" alt="#c13004"> 2: Unique patterns of the modules based on a range of minimal number of taxa per module. The darkened columns highlighted the duplicated patterns which are not included in the consensus-based evaluations.<br>
</li>
<li>
<img src="https://via.placeholder.com/15/820104/000000?text=+" alt="#820104"> 3: Manually select the pattern columns. Default will select all unique patterns with modules greater or equal to ten.</li>
<li>
<img src="https://via.placeholder.com/15/3e55a0/000000?text=+" alt="#3e55a0"> 4: Manually select the optimal number of modules based on the silhouette, consensus and correlation plots below.</li>
<li>
<img src="https://via.placeholder.com/15/3e55a0/000000?text=+" alt="#3e55a0"> 5: Generate the code the use should run after confirming the optimal selected patterns and optimal number of modules.</li>
<li>
<img src="https://via.placeholder.com/15/427f94/000000?text=+" alt="#427f94"> 6: Plot display panel including silhouette, consensus and correlation plots.</li>
<li><a href="https://zhoulabncsu.github.io/C3NA/articles/rMD4_ModEvals.html" class="external-link">Detailed Guide</a></li>
</ul>
</div>
<div class="section level3">
<h3 id="compare-two-phenotypes-shiny">Compare Two Phenotypes Shiny<a class="anchor" aria-label="anchor" href="#compare-two-phenotypes-shiny"></a>
</h3>
<p><img src="reference/figures/Shiny2PanelGuide.jpg" align="center"></p>
<ul>
<li>
<img src="https://via.placeholder.com/15/4d4d4d/000000?text=+" alt="#4d4d4d"> 1: Shiny panel navigation, including the ‘Compare Two Conditions’ and ‘Download Panel’.</li>
<li>
<img src="https://via.placeholder.com/15/c13004/000000?text=+" alt="#c13004"> 2: Phenotype-based information from the two conditions, including shared and different taxa from various taxonomic levels.</li>
<li>
<img src="https://via.placeholder.com/15/427f94/000000?text=+" alt="#427f94"> 3: Module-based taxa comparison between the two phenotypes.</li>
<li>
<img src="https://via.placeholder.com/15/13394e/000000?text=+" alt="#13394e"> 4: Module preservation investigation, the left plot shows the Median Rank Vs. Z Summary with the node size reflect number of taxa in the respective module. The right plot is an interactive table from react to the click of the module preservation plot.</li>
<li>
<img src="https://via.placeholder.com/15/9a0002/000000?text=+" alt="#9a0002"> 5: Taxa selector for the interactive display in Panels 6-8. The user can select multiple taxa from the left columns (includes all significant taxa from both phenotypes), to remove the selected taxon, you can click the taxon from the right column</li>
<li>
<img src="https://via.placeholder.com/15/5c0004/000000?text=+" alt="#5c0004"> 6: Bar plot for the taxa membership among the modules from either phenotypes. The user can switch between viewing all taxa or the selected taxa from the ‘Taxa Selector’. For better display, the bar plot will only display the select taxon and its children taxonomic level taxa. Please click on the left filter icon to view plot-based options.</li>
<li>
<img src="https://via.placeholder.com/15/359d8b/000000?text=+" alt="#359d8b"> 7: Network plot based on the selected taxa. Mulitple options are available from the top left filter icon.</li>
<li>
<img src="https://via.placeholder.com/15/005b5e/000000?text=+" alt="#005b5e"> 8: Single and sequentially clicked taxa and their respective log2-transformed count from both phenotypes.</li>
<li><a href="https://zhoulabncsu.github.io/C3NA/articles/rMD5_ComparePhenotypes.html" class="external-link">Detailed Guide</a></li>
</ul>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/zhouLabNCSU/C3NA/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/zhouLabNCSU/C3NA/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing C3NA</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Kuncheng Song <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-8680-1970" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Yi-Hui Zhou <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-4092-7463" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/zhouLabNCSU/C3NA/actions/workflows/check-standard.yaml" class="external-link"><img src="https://github.com/zhouLabNCSU/C3NA/actions/workflows/check-standard.yaml/badge.svg" alt="R-CMD-check (R V4.1)"></a></li>
<li><a href="https://github.com/zhouLabNCSU/C3NA/actions/workflows/pages/pages-build-deployment" class="external-link"><img src="https://github.com/zhouLabNCSU/C3NA/actions/workflows/pages/pages-build-deployment/badge.svg" alt="pkgdown-deployment"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Kuncheng Song, Yi-Hui Zhou.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
