<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Abundance Analysis â€¢ C3NA</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Differential Abundance Analysis">
<meta property="og:description" content="C3NA">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">C3NA</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/C3NA.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/rMD1_Demo.html">Tutorial with Demo</a>
    </li>
    <li>
      <a href="../articles/rMD2_DA.html">Differential Abundance Analysis</a>
    </li>
    <li>
      <a href="../articles/rMD3_FAQ.html">FAQ</a>
    </li>
    <li>
      <a href="../articles/rMD4_ModEvals.html">Tutorial on Using Module Evaluation Shiny</a>
    </li>
    <li>
      <a href="../articles/rMD5_ComparePhenotypes.html">Tutorials on Using Compare Phenotypes Shiny</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/zhouLabNCSU/C3NA/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="rMD2_DA_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="rMD2_DA_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="rMD2_DA_files/datatables-binding-0.20/datatables.js"></script><link href="rMD2_DA_files/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="rMD2_DA_files/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="rMD2_DA_files/dt-core-1.11.3/js/jquery.dataTables.min.js"></script><link href="rMD2_DA_files/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet">
<script src="rMD2_DA_files/crosstalk-1.2.0/js/crosstalk.min.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential Abundance Analysis</h1>
                        <h4 data-toc-skip class="author">Kuncheng Song</h4>
            
            <h4 data-toc-skip class="date">3/16/2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/zhouLabNCSU/C3NA/blob/HEAD/vignettes/rMD2_DA.Rmd" class="external-link"><code>vignettes/rMD2_DA.Rmd</code></a></small>
      <div class="hidden name"><code>rMD2_DA.Rmd</code></div>

    </div>

    
    
<p>This is a brief tutorial for conducting differential abundance analysis between two phenotypes and how to correctly add any types of DA analysis to C3NA object for interactive investigation of the co-occurence network. There are three differential abundance analysis done, including ANCOMBC, ALDEx2, and MaAsLin2, all of which conduct differential abundance analysis for each of the taxonomic levels individually. The results are summarized into a single DA table to be added to C3NA. As recommended by other studies, we will filter the count matrices from each taxonomic levels by 10% prevalance prior to running the DA methods.</p>
<div class="section level2">
<h2 id="library">Library<a class="anchor" aria-label="anchor" href="#library"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/FrederickHuangLin/ANCOMBC" class="external-link">ANCOMBC</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ggloor/ALDEx_bioc" class="external-link">ALDEx2</a></span><span class="op">)</span>
<span class="co">#&gt; Loading required package: zCompositions</span>
<span class="co">#&gt; Loading required package: MASS</span>
<span class="co">#&gt; Loading required package: NADA</span>
<span class="co">#&gt; Loading required package: survival</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'NADA'</span>
<span class="co">#&gt; The following object is masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     cor</span>
<span class="co">#&gt; Loading required package: truncnorm</span>
<span class="co">#&gt; Loading required package: Rfast</span>
<span class="co">#&gt; Loading required package: Rcpp</span>
<span class="co">#&gt; Loading required package: RcppZiggurat</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://huttenhower.sph.harvard.edu/maaslin2" class="external-link">Maaslin2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jeroen/curl" class="external-link">curl</a></span><span class="op">)</span>
<span class="co">#&gt; Using libcurl 7.64.1 with Schannel</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/DT" class="external-link">DT</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">phyloseq</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">metagMisc</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-1--loading-the-phyloseq-data-from-github">Step 1. Loading the phyloseq data from GitHub<a class="anchor" aria-label="anchor" href="#step-1--loading-the-phyloseq-data-from-github"></a>
</h2>
<p>The downloaded data is a phyloseq object with 261 samples, of which 127 are Colorectal Cancer Patients and 134 are healthy control.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">githubURL</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="st">"https://github.com/zhouLabNCSU/C3NA_ScriptsAndData/raw/main/RPackageTutorialData/Post-initiateC3NA/cancer_dada2_CancerAndNormal.rds"</span><span class="op">)</span>
<span class="va">CancerAndNormal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/connections.html" class="external-link">url</a></span><span class="op">(</span><span class="va">githubURL</span>, method<span class="op">=</span><span class="st">"libcurl"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">CancerAndNormal</span><span class="op">)</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 14635 taxa and 261 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 261 samples by 75 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 14635 taxa by 7 taxonomic ranks ]</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-2--generating-the-data-matrices-for-differential-abundance-analysis">Step 2. Generating the Data Matrices for Differential Abundance Analysis<a class="anchor" aria-label="anchor" href="#step-2--generating-the-data-matrices-for-differential-abundance-analysis"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove OTUs that do not present among these phenotypes and reformat the data</span>
<span class="va">newPS_otu</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">CancerAndNormal</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">newPS_otu</span> <span class="op">=</span> <span class="va">newPS_otu</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">newPS_otu</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">newPS_tax</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">CancerAndNormal</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">newPS_tax</span> <span class="op">=</span> <span class="va">newPS_tax</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">newPS_otu</span><span class="op">)</span>, <span class="op">]</span>
<span class="va">newPS_meta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">CancerAndNormal</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="va">newPS_meta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">newPS_meta</span>, <span class="va">diagnosis</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Normal"</span>, <span class="st">"Cancer"</span><span class="op">)</span><span class="op">)</span>
<span class="va">newPS_ps</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html" class="external-link">phyloseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">newPS_otu</span>, taxa_are_rows<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,
                             <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">newPS_tax</span><span class="op">)</span><span class="op">)</span>,
                             <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">newPS_meta</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">newPS_ps</span><span class="op">)</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 8055 taxa and 261 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 261 samples by 75 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 8055 taxa by 7 taxonomic ranks ]</span>
<span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">newPS_tax</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-e54e535c7c68dd2bd370" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e54e535c7c68dd2bd370">{"x":{"filter":"none","vertical":false,"data":[["OTU1","OTU2","OTU3","OTU4","OTU5","OTU6"],["Bacteria","Bacteria","Bacteria","Bacteria","Bacteria","Bacteria"],["Firmicutes","Bacteroidota","Firmicutes","Firmicutes","Verrucomicrobiota","Firmicutes"],["Clostridia","Bacteroidia","Clostridia","Clostridia","Verrucomicrobiae","Clostridia"],["Lachnospirales","Bacteroidales","Lachnospirales","Lachnospirales","Verrucomicrobiales","Lachnospirales"],["Lachnospiraceae","Bacteroidaceae","Lachnospiraceae","Lachnospiraceae","Akkermansiaceae","Lachnospiraceae"],["Blautia","Bacteroides","Agathobacter","Blautia","Akkermansia","Anaerostipes"],[null,"Bacteroides_vulgatus",null,null,null,null]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Domain<\/th>\n      <th>Phylum<\/th>\n      <th>Class<\/th>\n      <th>Order<\/th>\n      <th>Family<\/th>\n      <th>Genus<\/th>\n      <th>Species<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script><p><em>Ideally, this should match the taxa names used for the C3NA. Ideally, there should not be any t</em></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Create a phyloseq object for each taxonomic level</span>
<span class="va">taxaLvls</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Phylum"</span>, <span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span><span class="op">)</span>
<span class="va">taxaHeaders</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"p"</span>, <span class="st">"c"</span>, <span class="st">"o"</span>, <span class="st">"f"</span>, <span class="st">"g"</span>, <span class="st">"s"</span><span class="op">)</span>
<span class="va">taxaCountMatrices_Abs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">psList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span><span class="op">(</span><span class="va">t</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="va">curLevel</span> <span class="op">=</span> <span class="va">taxaLvls</span><span class="op">[</span><span class="va">t</span><span class="op">]</span>
  <span class="va">psTemp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_glom.html" class="external-link">tax_glom</a></span><span class="op">(</span><span class="va">newPS_ps</span>, <span class="va">curLevel</span>, NArm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
  <span class="va">curOTUs</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">psTemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
  <span class="co">## Summing NAs</span>
  <span class="va">curOTUs</span><span class="op">$</span><span class="va">TEMP</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">psTemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="va">curLevel</span><span class="op">]</span><span class="op">)</span>
  <span class="va">curOTUs</span> <span class="op">&lt;-</span> <span class="va">curOTUs</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>TEMP <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">TEMP</span><span class="op">)</span>, <span class="st">"NA"</span>, <span class="va">TEMP</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">TEMP</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/summarise_all.html" class="external-link">summarise_if</a></span><span class="op">(</span><span class="va">is.numeric</span>, <span class="va">sum</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html" class="external-link">column_to_rownames</a></span><span class="op">(</span><span class="st">"TEMP"</span><span class="op">)</span>
  <span class="co">## Add the prefix of taxa</span>
  <span class="va">curHeader</span> <span class="op">=</span> <span class="va">taxaHeaders</span><span class="op">[</span><span class="va">t</span><span class="op">]</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">curOTUs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">curHeader</span>,<span class="st">"_"</span>,<span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">curOTUs</span><span class="op">)</span><span class="op">)</span>

  <span class="co">## Phyloseq Object</span>
  <span class="va">curData</span> <span class="op">=</span> <span class="va">curOTUs</span>
  <span class="va">curData</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">curData</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">0</span>
  <span class="va">curData</span> <span class="op">&lt;-</span> <span class="va">curData</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">curData</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span>, <span class="op">]</span>  
  <span class="va">tempTaxa</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">curData</span><span class="op">)</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">tempTaxa</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"MixedTaxa"</span>
  <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tempTaxa</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tempTaxa</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>
  <span class="va">tempPS</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/phyloseq.html" class="external-link">phyloseq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">curData</span>, taxa_are_rows<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,
                    <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html" class="external-link">tax_table</a></span><span class="op">(</span><span class="va">tempTaxa</span><span class="op">)</span>,
                    <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">psTemp</span><span class="op">)</span><span class="op">)</span>

  <span class="co">## Save the Objects</span>
  <span class="va">taxaCountMatrices_Abs</span><span class="op">[[</span><span class="va">curLevel</span><span class="op">]</span><span class="op">]</span> <span class="op">=</span> <span class="va">curOTUs</span>
  <span class="va">psList</span><span class="op">[[</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">t</span><span class="op">]</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">tempPS</span>
<span class="op">}</span></code></pre></div>
<div class="section level4">
<h4 id="create-a-da-summary-table">Create a DA Summary Table<a class="anchor" aria-label="anchor" href="#create-a-da-summary-table"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">DA_Summary</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
  TaxaName <span class="op">=</span> <span class="cn">NULL</span>, DAMethod <span class="op">=</span> <span class="cn">NULL</span>, DiffAbn <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-3--ancombc">Step 3. ANCOMBC<a class="anchor" aria-label="anchor" href="#step-3--ancombc"></a>
</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
  <span class="va">curSample</span> <span class="op">=</span> <span class="va">psList</span><span class="op">[[</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>
  <span class="va">tempOriNTaxa</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">curSample</span><span class="op">)</span><span class="op">)</span>
  <span class="va">curSample</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/metagMisc/man/phyloseq_filter_prevalence.html" class="external-link">phyloseq_filter_prevalence</a></span><span class="op">(</span><span class="va">curSample</span>, prev.trh <span class="op">=</span> <span class="fl">0.1</span>, abund.trh <span class="op">=</span> <span class="cn">NULL</span>,
    threshold_condition <span class="op">=</span> <span class="st">"AND"</span>, abund.type <span class="op">=</span> <span class="st">"total"</span><span class="op">)</span>
  <span class="co">## ANCOMBC with BH</span>
  <span class="va">curANCOMBC</span> <span class="op">=</span> <span class="fu">ANCOMBC</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ANCOMBC/man/ancombc.html" class="external-link">ancombc</a></span><span class="op">(</span>phyloseq <span class="op">=</span> <span class="va">curSample</span>, formula <span class="op">=</span> <span class="st">"diagnosis"</span>,
                                p_adj_method <span class="op">=</span> <span class="st">"BH"</span>, zero_cut <span class="op">=</span> <span class="fl">1</span>, 
                                lib_cut <span class="op">=</span> <span class="fl">0</span>, group <span class="op">=</span> <span class="st">"diagnosis"</span>,
                                struc_zero <span class="op">=</span> <span class="cn">TRUE</span>, neg_lb <span class="op">=</span> <span class="cn">TRUE</span>, 
                                tol <span class="op">=</span> <span class="fl">1e-5</span>, max_iter <span class="op">=</span> <span class="fl">100</span>, conserve <span class="op">=</span> <span class="cn">FALSE</span>, 
                                alpha <span class="op">=</span> <span class="fl">0.05</span>, global <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
  <span class="va">curANCOMBC_Res</span> <span class="op">=</span> <span class="va">curANCOMBC</span><span class="op">$</span><span class="va">res</span>
  <span class="va">DA_Summary_Summary_temp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
      DAMethod <span class="op">=</span> <span class="st">"ANCOMBC"</span>,
      TaxaName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">curANCOMBC_Res</span><span class="op">$</span><span class="va">beta</span><span class="op">)</span>,
      diff_abn <span class="op">=</span> <span class="va">curANCOMBC_Res</span><span class="op">$</span><span class="va">diff_abn</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span><span class="op">)</span>
  <span class="va">DA_Summary</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">DA_Summary</span>, <span class="va">DA_Summary_Summary_temp</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">#&gt; [1] "Phylum"</span>
<span class="co">#&gt; Warning: The multi-group comparison will be deactivated as the group variable</span>
<span class="co">#&gt; has &lt; 3 categories</span>
<span class="co">#&gt; [1] "Class"</span>
<span class="co">#&gt; Warning: The multi-group comparison will be deactivated as the group variable</span>
<span class="co">#&gt; has &lt; 3 categories</span>
<span class="co">#&gt; [1] "Order"</span>
<span class="co">#&gt; Warning: The multi-group comparison will be deactivated as the group variable</span>
<span class="co">#&gt; has &lt; 3 categories</span>
<span class="co">#&gt; [1] "Family"</span>
<span class="co">#&gt; Warning: The multi-group comparison will be deactivated as the group variable</span>
<span class="co">#&gt; has &lt; 3 categories</span>
<span class="co">#&gt; [1] "Genus"</span>
<span class="co">#&gt; Warning: The multi-group comparison will be deactivated as the group variable</span>
<span class="co">#&gt; has &lt; 3 categories</span>
<span class="co">#&gt; [1] "Species"</span>
<span class="co">#&gt; Warning: The multi-group comparison will be deactivated as the group variable</span>
<span class="co">#&gt; has &lt; 3 categories</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-4--aldex2">Step 4. ALDEx2<a class="anchor" aria-label="anchor" href="#step-4--aldex2"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
  <span class="va">curSample</span> <span class="op">=</span> <span class="va">psList</span><span class="op">[[</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>
  <span class="va">tempOriNTaxa</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">curSample</span><span class="op">)</span><span class="op">)</span>
  <span class="va">curSample</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/metagMisc/man/phyloseq_filter_prevalence.html" class="external-link">phyloseq_filter_prevalence</a></span><span class="op">(</span><span class="va">curSample</span>, prev.trh <span class="op">=</span> <span class="fl">0.1</span>, abund.trh <span class="op">=</span> <span class="cn">NULL</span>,
    threshold_condition <span class="op">=</span> <span class="st">"AND"</span>, abund.type <span class="op">=</span> <span class="st">"total"</span><span class="op">)</span>
  <span class="co">## ALDEx2 with BH</span>
  <span class="va">curALDEx2</span> <span class="op">&lt;-</span> <span class="fu">ALDEx2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ALDEx2/man/aldex.html" class="external-link">aldex</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">curSample</span><span class="op">)</span><span class="op">)</span>, 
                             <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">curSample</span><span class="op">)</span><span class="op">$</span><span class="va">diagnosis</span>, 
                             test<span class="op">=</span><span class="st">"t"</span>, effect <span class="op">=</span> <span class="cn">TRUE</span>, denom<span class="op">=</span><span class="st">"all"</span><span class="op">)</span>
  <span class="va">DA_Summary_Summary_temp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
    DAMethod <span class="op">=</span> <span class="st">"ALDEx2"</span>,
    TaxaName <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/tibble/man/rownames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">curALDEx2</span><span class="op">)</span>,
    diff_abn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">curALDEx2</span><span class="op">$</span><span class="va">wi.eBH</span> <span class="op">&lt;=</span><span class="fl">0.05</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span>
  <span class="va">DA_Summary</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">DA_Summary</span>, <span class="va">DA_Summary_Summary_temp</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">#&gt; [1] "Phylum"</span>
<span class="co">#&gt; aldex.clr: generating Monte-Carlo instances and clr values</span>
<span class="co">#&gt; operating in serial mode</span>
<span class="co">#&gt; computing center with all features</span>
<span class="co">#&gt; aldex.ttest: doing t-test</span>
<span class="co">#&gt; aldex.effect: calculating effect sizes</span>
<span class="co">#&gt; [1] "Class"</span>
<span class="co">#&gt; aldex.clr: generating Monte-Carlo instances and clr values</span>
<span class="co">#&gt; operating in serial mode</span>
<span class="co">#&gt; computing center with all features</span>
<span class="co">#&gt; aldex.ttest: doing t-test</span>
<span class="co">#&gt; aldex.effect: calculating effect sizes</span>
<span class="co">#&gt; [1] "Order"</span>
<span class="co">#&gt; aldex.clr: generating Monte-Carlo instances and clr values</span>
<span class="co">#&gt; operating in serial mode</span>
<span class="co">#&gt; computing center with all features</span>
<span class="co">#&gt; aldex.ttest: doing t-test</span>
<span class="co">#&gt; aldex.effect: calculating effect sizes</span>
<span class="co">#&gt; [1] "Family"</span>
<span class="co">#&gt; aldex.clr: generating Monte-Carlo instances and clr values</span>
<span class="co">#&gt; operating in serial mode</span>
<span class="co">#&gt; computing center with all features</span>
<span class="co">#&gt; aldex.ttest: doing t-test</span>
<span class="co">#&gt; aldex.effect: calculating effect sizes</span>
<span class="co">#&gt; [1] "Genus"</span>
<span class="co">#&gt; aldex.clr: generating Monte-Carlo instances and clr values</span>
<span class="co">#&gt; operating in serial mode</span>
<span class="co">#&gt; computing center with all features</span>
<span class="co">#&gt; aldex.ttest: doing t-test</span>
<span class="co">#&gt; aldex.effect: calculating effect sizes</span>
<span class="co">#&gt; [1] "Species"</span>
<span class="co">#&gt; aldex.clr: generating Monte-Carlo instances and clr values</span>
<span class="co">#&gt; operating in serial mode</span>
<span class="co">#&gt; computing center with all features</span>
<span class="co">#&gt; aldex.ttest: doing t-test</span>
<span class="co">#&gt; aldex.effect: calculating effect sizes</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-5--maaslin2">Step 5. MaAsLin2<a class="anchor" aria-label="anchor" href="#step-5--maaslin2"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">outputFileNames</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span>
<span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
  <span class="va">curSample</span> <span class="op">=</span> <span class="va">psList</span><span class="op">[[</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">]</span>
  <span class="va">tempOriNTaxa</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">curSample</span><span class="op">)</span><span class="op">)</span>
  <span class="va">curSample</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/metagMisc/man/phyloseq_filter_prevalence.html" class="external-link">phyloseq_filter_prevalence</a></span><span class="op">(</span><span class="va">curSample</span>, prev.trh <span class="op">=</span> <span class="fl">0.1</span>, abund.trh <span class="op">=</span> <span class="cn">NULL</span>,
    threshold_condition <span class="op">=</span> <span class="st">"AND"</span>, abund.type <span class="op">=</span> <span class="st">"total"</span><span class="op">)</span>
  <span class="co">## MaAsLin2 with BH</span>
  <span class="va">curMaAsLin2</span> <span class="op">=</span> <span class="fu">Maaslin2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Maaslin2/man/Maaslin2.html" class="external-link">Maaslin2</a></span><span class="op">(</span>
    input_data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html" class="external-link">otu_table</a></span><span class="op">(</span><span class="va">curSample</span><span class="op">)</span><span class="op">)</span>, 
    input_metadata <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html" class="external-link">sample_data</a></span><span class="op">(</span><span class="va">curSample</span><span class="op">)</span><span class="op">)</span>, 
    output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">)</span>, 
    transform <span class="op">=</span> <span class="st">"AST"</span>,
    fixed_effects <span class="op">=</span> <span class="st">"diagnosis"</span>,
    normalization <span class="op">=</span> <span class="st">"TSS"</span>,
    standardize <span class="op">=</span> <span class="cn">FALSE</span>,
    min_prevalence <span class="op">=</span> <span class="fl">0</span>, plot_heatmap <span class="op">=</span> <span class="cn">FALSE</span>, plot_scatter <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
  <span class="va">DA_Summary_Summary_temp</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>
    DAMethod <span class="op">=</span> <span class="st">"MaAsLin2"</span>,
    TaxaName <span class="op">=</span> <span class="va">curMaAsLin2</span><span class="op">$</span><span class="va">results</span><span class="op">$</span><span class="va">feature</span>,
    diff_abn <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">curMaAsLin2</span><span class="op">$</span><span class="va">results</span><span class="op">$</span><span class="va">qval</span> <span class="op">&lt;=</span><span class="fl">0.05</span>, <span class="cn">TRUE</span>, <span class="cn">FALSE</span><span class="op">)</span>
  <span class="op">)</span>
  <span class="va">DA_Summary</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">DA_Summary</span>, <span class="va">DA_Summary_Summary_temp</span><span class="op">)</span>
  <span class="co">## Delete the temp folder </span>
  <span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">taxaLvls</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>,recursive<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="op">}</span>
<span class="co">#&gt; [1] "Phylum"</span>
<span class="co">#&gt; [1] "Creating output folder"</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Writing function arguments to log file</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Verifying options selected are valid</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Determining format of input files</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Input format is data samples as columns and metadata samples as rows</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Formula for fixed effects: expr ~  diagnosis</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Filter data based on min abundance and min prevalence</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Total samples in data: 261</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Min samples required with min abundance for a feature not to be filtered: 0.000000</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Total filtered features: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Filtered feature names from abundance and prevalence filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Total filtered features with variance filtering: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Filtered feature names from variance filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Running selected normalization method: TSS</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Bypass z-score application to metadata</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Running selected transform method: AST</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Running selected analysis method: LM</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 1, p_Actinobacteriota</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 2, p_Bacteroidota</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 3, p_Campilobacterota</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 4, p_Cyanobacteria</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 5, p_Desulfobacterota</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 6, p_Euryarchaeota</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 7, p_Firmicutes</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 8, p_Fusobacteriota</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 9, p_Proteobacteria</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 10, p_Synergistota</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Fitting model to feature number 11, p_Verrucomicrobiota</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Counting total values for each feature</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Writing residuals to file Phylum/residuals.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Writing fitted values to file Phylum/fitted.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Writing all results to file (ordered by increasing q-values): Phylum/all_results.tsv</span>
<span class="co">#&gt; 2022-03-20 18:54:47 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): Phylum/significant_results.tsv</span>
<span class="co">#&gt; [1] "Class"</span>
<span class="co">#&gt; [1] "Creating output folder"</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Writing function arguments to log file</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Verifying options selected are valid</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Determining format of input files</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Input format is data samples as columns and metadata samples as rows</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Formula for fixed effects: expr ~  diagnosis</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Filter data based on min abundance and min prevalence</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Total samples in data: 261</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Min samples required with min abundance for a feature not to be filtered: 0.000000</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Total filtered features: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Filtered feature names from abundance and prevalence filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Total filtered features with variance filtering: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Filtered feature names from variance filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Running selected normalization method: TSS</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Bypass z-score application to metadata</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Running selected transform method: AST</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Running selected analysis method: LM</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 1, c_Actinobacteria</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 2, c_Alphaproteobacteria</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 3, c_Bacilli</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 4, c_Bacteroidia</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 5, c_Campylobacteria</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 6, c_Clostridia</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 7, c_Coriobacteriia</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 8, c_Cyanobacteriia</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 9, c_Desulfovibrionia</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 10, c_Fusobacteriia</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 11, c_Gammaproteobacteria</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 12, c_Incertae_Sedis</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 13, c_Methanobacteria</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 14, c_NA</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 15, c_Negativicutes</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 16, c_Synergistia</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 17, c_uncultured</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 18, c_Vampirivibrionia</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Fitting model to feature number 19, c_Verrucomicrobiae</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Counting total values for each feature</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Writing residuals to file Class/residuals.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Writing fitted values to file Class/fitted.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Writing all results to file (ordered by increasing q-values): Class/all_results.tsv</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): Class/significant_results.tsv</span>
<span class="co">#&gt; [1] "Order"</span>
<span class="co">#&gt; [1] "Creating output folder"</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Writing function arguments to log file</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Verifying options selected are valid</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Determining format of input files</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Input format is data samples as columns and metadata samples as rows</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Formula for fixed effects: expr ~  diagnosis</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Filter data based on min abundance and min prevalence</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Total samples in data: 261</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Min samples required with min abundance for a feature not to be filtered: 0.000000</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Total filtered features: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:48 INFO::Filtered feature names from abundance and prevalence filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Total filtered features with variance filtering: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Filtered feature names from variance filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Running selected normalization method: TSS</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Bypass z-score application to metadata</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Running selected transform method: AST</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Running selected analysis method: LM</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 1, o_Acidaminococcales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 2, o_Actinomycetales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 3, o_Bacteroidales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 4, o_Bifidobacteriales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 5, o_Burkholderiales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 6, o_Campylobacterales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 7, o_Chloroplast</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 8, o_Christensenellales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 9, o_Clostridia</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 10, o_Clostridia_UCG.014</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 11, o_Clostridia_vadinBB60_group</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 12, o_Clostridiales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 13, o_Coriobacteriales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 14, o_Corynebacteriales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 15, o_Desulfovibrionales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 16, o_DTU014</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 17, o_Enterobacterales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 18, o_Erysipelotrichales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 19, o_Eubacteriales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 20, o_Flavobacteriales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 21, o_Fusobacteriales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 22, o_Gastranaerophilales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 23, o_Izemoplasmatales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 24, o_Lachnospirales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 25, o_Lactobacillales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 26, o_Methanobacteriales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 27, o_Micrococcales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 28, o_Monoglobales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 29, o_NA</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 30, o_Oscillospirales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 31, o_Paenibacillales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 32, o_Pasteurellales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 33, o_Peptococcales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 34, o_Peptostreptococcales.Tissierellales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 35, o_Pseudomonadales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 36, o_RF39</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 37, o_Rhizobiales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 38, o_Rhodospirillales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 39, o_Staphylococcales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 40, o_Synergistales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 41, o_uncultured</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 42, o_Veillonellales.Selenomonadales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 43, o_Verrucomicrobiales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Counting total values for each feature</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing residuals to file Order/residuals.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing fitted values to file Order/fitted.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing all results to file (ordered by increasing q-values): Order/all_results.tsv</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): Order/significant_results.tsv</span>
<span class="co">#&gt; [1] "Family"</span>
<span class="co">#&gt; [1] "Creating output folder"</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing function arguments to log file</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Verifying options selected are valid</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Determining format of input files</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Input format is data samples as columns and metadata samples as rows</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Formula for fixed effects: expr ~  diagnosis</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Filter data based on min abundance and min prevalence</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Total samples in data: 261</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Min samples required with min abundance for a feature not to be filtered: 0.000000</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Total filtered features: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Filtered feature names from abundance and prevalence filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Total filtered features with variance filtering: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Filtered feature names from variance filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Running selected normalization method: TSS</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Bypass z-score application to metadata</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Running selected transform method: AST</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Running selected analysis method: LM</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 1, f_.Clostridium._methylpentosum_group</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 2, f_.Eubacterium._coprostanoligenes_group</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 3, f_Acidaminococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 4, f_Actinomycetaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 5, f_Akkermansiaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 6, f_Anaerofustaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 7, f_Anaerovoracaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 8, f_Atopobiaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 9, f_Bacteroidaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 10, f_Barnesiellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 11, f_Beijerinckiaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 12, f_Bifidobacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 13, f_Butyricicoccaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 14, f_Campylobacteraceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 15, f_Carnobacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 16, f_Chloroplast</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 17, f_Christensenellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 18, f_Clostridia_UCG.014</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 19, f_Clostridia_vadinBB60_group</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 20, f_Clostridiaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 21, f_Coriobacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 22, f_Coriobacteriales_Incertae_Sedis</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 23, f_Corynebacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 24, f_Defluviitaleaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 25, f_Desulfovibrionaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 26, f_DTU014</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 27, f_Eggerthellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 28, f_Enterobacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 29, f_Enterococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 30, f_Erysipelatoclostridiaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 31, f_Erysipelotrichaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 32, f_Eubacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 33, f_Flavobacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 34, f_Fusobacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 35, f_Gastranaerophilales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 36, f_Gemellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 37, f_Hungateiclostridiaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 38, f_Izemoplasmatales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 39, f_Lachnospiraceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 40, f_Lactobacillaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 41, f_Leuconostocaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 42, f_Marinifilaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 43, f_Methanobacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 44, f_Micrococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 45, f_Monoglobaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 46, f_Muribaculaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 47, f_NA</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 48, f_Neisseriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 49, f_Oscillospiraceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 50, f_Oscillospirales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 51, f_Oxalobacteraceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 52, f_Paenibacillaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 53, f_Pasteurellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 54, f_Peptococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 55, f_Peptostreptococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 56, f_Peptostreptococcales.Tissierellales</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 57, f_Porphyromonadaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 58, f_Prevotellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 59, f_Pseudomonadaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 60, f_RF39</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 61, f_Rikenellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 62, f_Ruminococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 63, f_Selenomonadaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 64, f_Streptococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 65, f_Sutterellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 66, f_Synergistaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 67, f_Tannerellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 68, f_UCG.010</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 69, f_uncultured</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Fitting model to feature number 70, f_Veillonellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Counting total values for each feature</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing residuals to file Family/residuals.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing fitted values to file Family/fitted.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing all results to file (ordered by increasing q-values): Family/all_results.tsv</span>
<span class="co">#&gt; 2022-03-20 18:54:49 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): Family/significant_results.tsv</span>
<span class="co">#&gt; [1] "Genus"</span>
<span class="co">#&gt; [1] "Creating output folder"</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Writing function arguments to log file</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Verifying options selected are valid</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Determining format of input files</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Input format is data samples as columns and metadata samples as rows</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Formula for fixed effects: expr ~  diagnosis</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Filter data based on min abundance and min prevalence</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Total samples in data: 261</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Min samples required with min abundance for a feature not to be filtered: 0.000000</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Total filtered features: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Filtered feature names from abundance and prevalence filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Total filtered features with variance filtering: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Filtered feature names from variance filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Running selected normalization method: TSS</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Bypass z-score application to metadata</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Running selected transform method: AST</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Running selected analysis method: LM</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 1, g_.Clostridium._innocuum_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 2, g_.Clostridium._methylpentosum_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 3, g_.Eubacterium._brachy_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 4, g_.Eubacterium._coprostanoligenes_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 5, g_.Eubacterium._eligens_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 6, g_.Eubacterium._fissicatena_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 7, g_.Eubacterium._hallii_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 8, g_.Eubacterium._nodatum_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 9, g_.Eubacterium._ruminantium_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 10, g_.Eubacterium._siraeum_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 11, g_.Eubacterium._ventriosum_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 12, g_.Eubacterium._xylanophilum_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 13, g_.Ruminococcus._gauvreauii_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 14, g_.Ruminococcus._gnavus_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 15, g_.Ruminococcus._torques_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 16, g_Acidaminococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 17, g_Actinomyces</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 18, g_Adlercreutzia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 19, g_Agathobacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 20, g_Akkermansia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 21, g_Alistipes</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 22, g_Alloprevotella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 23, g_Anaerococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 24, g_Anaerofilum</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 25, g_Anaerofustis</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 26, g_Anaerosporobacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 27, g_Anaerostipes</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 28, g_Anaerotruncus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 29, g_Angelakisella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 30, g_Atopobium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 31, g_Bacteroides</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 32, g_Barnesiella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 33, g_Bifidobacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 34, g_Bilophila</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 35, g_Blautia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 36, g_Butyricicoccus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 37, g_Butyricimonas</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 38, g_Butyrivibrio</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 39, g_CAG.352</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 40, g_CAG.56</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 41, g_Campylobacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 42, g_Candidatus_Soleaferrea</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 43, g_Candidatus_Stoquefichus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 44, g_Catenibacillus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 45, g_Catenibacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 46, g_CHKCI002</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 47, g_Chloroplast</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 48, g_Christensenella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 49, g_Christensenellaceae_R.7_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 50, g_Cloacibacillus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 51, g_Clostridia_UCG.014</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 52, g_Clostridia_vadinBB60_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 53, g_Clostridium_sensu_stricto_1</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 54, g_Colidextribacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 55, g_Collinsella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 56, g_Coprobacillus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 57, g_Coprobacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 58, g_Coprococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 59, g_Corynebacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 60, g_Defluviitaleaceae_UCG.011</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 61, g_Desulfovibrio</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 62, g_Dialister</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 63, g_Dielma</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 64, g_Dorea</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 65, g_DTU014</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 66, g_DTU089</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 67, g_Eggerthella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 68, g_Eisenbergiella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 69, g_Enterococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 70, g_Enterorhabdus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 71, g_Erysipelatoclostridium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 72, g_Erysipelotrichaceae_UCG.003</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 73, g_Escherichia.Shigella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 74, g_Eubacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 75, g_Faecalibacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 76, g_Faecalitalea</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 77, g_Family_XIII_AD3011_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 78, g_Family_XIII_UCG.001</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 79, g_Flavonifractor</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 80, g_Fournierella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 81, g_Frisingicoccus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 82, g_Fusicatenibacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 83, g_Fusobacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 84, g_Gastranaerophilales</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 85, g_GCA.900066575</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 86, g_GCA.900066755</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 87, g_Gemella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 88, g_Gordonibacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 89, g_Granulicatella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 90, g_Haemophilus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 91, g_Holdemanella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 92, g_Holdemania</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 93, g_Howardella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 94, g_Hungatella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 95, g_Hydrogenoanaerobacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 96, g_Incertae_Sedis</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 97, g_Intestinibacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 98, g_Intestinimonas</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 99, g_Izemoplasmatales</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 100, g_Lachnoclostridium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 101, g_Lachnospira</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 102, g_Lachnospiraceae_FCS020_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 103, g_Lachnospiraceae_ND3007_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 104, g_Lachnospiraceae_NK4A136_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 105, g_Lachnospiraceae_NK4B4_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 106, g_Lachnospiraceae_UCG.001</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 107, g_Lachnospiraceae_UCG.004</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 108, g_Lachnospiraceae_UCG.008</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 109, g_Lachnospiraceae_UCG.009</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 110, g_Lachnospiraceae_UCG.010</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 111, g_Lactobacillus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 112, g_Lactococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 113, g_Lactonifactor</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 114, g_Leuconostoc</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 115, g_Marvinbryantia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 116, g_Megasphaera</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 117, g_Merdibacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 118, g_Methanobrevibacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 119, g_Methylobacterium.Methylorubrum</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 120, g_Mogibacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 121, g_Monoglobus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 122, g_Moryella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 123, g_Muribaculaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 124, g_NA</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 125, g_Negativibacillus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 126, g_Neisseria</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 127, g_NK4A214_group</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 128, g_Odoribacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 129, g_Oscillibacter</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 130, g_Oscillospira</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 131, g_Paenibacillus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 132, g_Paludicola</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 133, g_Parabacteroides</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 134, g_Paraprevotella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 135, g_Parasutterella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 136, g_Parvimonas</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 137, g_Peptococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 138, g_Peptoniphilus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 139, g_Peptostreptococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 140, g_Phascolarctobacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 141, g_Phocea</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 142, g_Porphyromonas</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 143, g_Prevotella</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 144, g_Prevotellaceae_UCG.001</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 145, g_Pseudomonas</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 146, g_RF39</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 147, g_Romboutsia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 148, g_Roseburia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 149, g_Rothia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 150, g_Ruminiclostridium</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 151, g_Ruminococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 152, g_S5.A14a</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 153, g_Sellimonas</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 154, g_Senegalimassilia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 155, g_Shuttleworthia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 156, g_Slackia</span>
<span class="co">#&gt; 2022-03-20 18:54:50 INFO::Fitting model to feature number 157, g_Solobacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 158, g_Streptococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 159, g_Subdoligranulum</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 160, g_Sutterella</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 161, g_Terrisporobacter</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 162, g_Turicibacter</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 163, g_Tyzzerella</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 164, g_UBA1819</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 165, g_UC5.1.2E3</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 166, g_UCG.002</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 167, g_UCG.003</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 168, g_UCG.005</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 169, g_UCG.008</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 170, g_UCG.009</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 171, g_UCG.010</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 172, g_uncultured</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 173, g_Veillonella</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Counting total values for each feature</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Writing residuals to file Genus/residuals.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Writing fitted values to file Genus/fitted.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Writing all results to file (ordered by increasing q-values): Genus/all_results.tsv</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): Genus/significant_results.tsv</span>
<span class="co">#&gt; [1] "Species"</span>
<span class="co">#&gt; [1] "Creating output folder"</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Writing function arguments to log file</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Verifying options selected are valid</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Determining format of input files</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Input format is data samples as columns and metadata samples as rows</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Formula for fixed effects: expr ~  diagnosis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Filter data based on min abundance and min prevalence</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Total samples in data: 261</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Min samples required with min abundance for a feature not to be filtered: 0.000000</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Total filtered features: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Filtered feature names from abundance and prevalence filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Total filtered features with variance filtering: 0</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Filtered feature names from variance filtering:</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Running selected normalization method: TSS</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Bypass z-score application to metadata</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Running selected transform method: AST</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Running selected analysis method: LM</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 1, s_.Clostridium._leptum</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 2, s_.Clostridium._scindens</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 3, s_.Clostridium._spiroforme</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 4, s_.Eubacterium._siraeum</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 5, s_Absiella_dolichum</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 6, s_Actinomyces_graevenitzii</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 7, s_Alistipes_indistinctus</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 8, s_Alistipes_inops</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 9, s_Alistipes_obesi</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 10, s_Alistipes_shahii</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 11, s_Alistipes_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 12, s_Anaerofustis_stercorihominis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 13, s_Anaerotruncus_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 14, s_bacterium_YE57</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 15, s_Bacteroides_eggerthii</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 16, s_Bacteroides_intestinalis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 17, s_Bacteroides_massiliensis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 18, s_Bacteroides_ovatus</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 19, s_Bacteroides_plebeius</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 20, s_Bacteroides_salyersiae</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 21, s_Bacteroides_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 22, s_Bacteroides_stercoris</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 23, s_Bacteroides_vulgatus</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 24, s_Blautia_hydrogenotrophica</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 25, s_Blautia_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 26, s_Christensenella_massiliensis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 27, s_Christensenella_minuta</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 28, s_Christensenella_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 29, s_Cloacibacillus_evryensis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 30, s_Clostridiales_bacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 31, s_Clostridium_perfringens</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 32, s_Clostridium_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 33, s_Colidextribacter_massiliensis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 34, s_Collinsella_stercoris</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 35, s_Coprococcus_eutactus</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 36, s_Corynebacterium_durum</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 37, s_Desulfovibrio_desulfuricans</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 38, s_Eubacterium_ramulus</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 39, s_Faecalitalea_cylindroides</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 40, s_gut_metagenome</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 41, s_human_gut</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 42, s_Lachnospiraceae_bacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 43, s_Lactococcus_lactis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 44, s_Massiliomicrobiota_timonensis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 45, s_Massiliprevotella_massiliensis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 46, s_metagenome</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 47, s_Mogibacterium_diversum</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 48, s_Mordavella_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 49, s_NA</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 50, s_Parabacteroides_distasonis</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 51, s_Parabacteroides_johnsonii</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 52, s_Parabacteroides_merdae</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 53, s_Porphyromonas_asaccharolytica</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 54, s_Ruminococcaceae_bacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 55, s_Ruminococcus_bicirculans</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 56, s_Ruminococcus_lactaris</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 57, s_Ruminococcus_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 58, s_Schaalia_odontolytica</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 59, s_Solobacterium_moorei</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 60, s_Streptococcus_mutans</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 61, s_Streptococcus_salivarius</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 62, s_Tyzzerella_sp.</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 63, s_uncultured_Alistipes</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 64, s_uncultured_Anaerovorax</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 65, s_uncultured_bacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 66, s_uncultured_Bacteroidales</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 67, s_uncultured_Blautia</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 68, s_uncultured_Clostridia</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 69, s_uncultured_Clostridiaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 70, s_uncultured_Clostridiales</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 71, s_uncultured_Clostridium</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 72, s_uncultured_Dorea</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 73, s_uncultured_Eubacteriaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 74, s_uncultured_Eubacterium</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 75, s_uncultured_Firmicutes</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 76, s_uncultured_Lachnospiraceae</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 77, s_uncultured_marine</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 78, s_uncultured_organism</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 79, s_uncultured_Peptococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 80, s_uncultured_prokaryote</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 81, s_uncultured_Roseburia</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 82, s_uncultured_rumen</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 83, s_uncultured_Ruminococcaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 84, s_uncultured_Ruminococcus</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 85, s_uncultured_Thermoanaerobacterales</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 86, s_uncultured_Veillonellaceae</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Fitting model to feature number 87, s_unidentified</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Counting total values for each feature</span>
<span class="co">#&gt; 2022-03-20 18:54:51 INFO::Writing residuals to file Species/residuals.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:52 INFO::Writing fitted values to file Species/fitted.rds</span>
<span class="co">#&gt; 2022-03-20 18:54:52 INFO::Writing all results to file (ordered by increasing q-values): Species/all_results.tsv</span>
<span class="co">#&gt; 2022-03-20 18:54:52 INFO::Writing the significant results (those which are less than or equal to the threshold of 0.250000 ) to file (ordered by increasing q-values): Species/significant_results.tsv</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="step-6--convert-the-differential-abundance-analysis-to-a-summary">Step 6. Convert the Differential Abundance Analysis to a Summary<a class="anchor" aria-label="anchor" href="#step-6--convert-the-differential-abundance-analysis-to-a-summary"></a>
</h2>
<p>The shiny application accept a data.frame as shown below to be included into the interactive network display.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">DASummaryTable</span> <span class="op">=</span> <span class="va">DA_Summary</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/tidyr/man/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>values_from <span class="op">=</span> <span class="va">diff_abn</span>, names_from <span class="op">=</span> <span class="va">DAMethod</span>, id_cols <span class="op">=</span> <span class="va">TaxaName</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/DT/man/DT-imports.html" class="external-link">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/pkg/dplyr/man/relocate.html" class="external-link">relocate</a></span><span class="op">(</span><span class="va">TaxaName</span><span class="op">)</span> <span class="co">## Relocate the TaxaName column to the front &lt;-- Important!</span>
<span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">DASummaryTable</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>pageLength <span class="op">=</span> <span class="fl">20</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-3e40670a14febb31556c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3e40670a14febb31556c">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455"],["p_Actinobacteriota","p_Bacteroidota","p_Campilobacterota","p_Cyanobacteria","p_Desulfobacterota","p_Euryarchaeota","p_Firmicutes","p_Fusobacteriota","p_Proteobacteria","p_Synergistota","p_Verrucomicrobiota","c_Actinobacteria","c_Alphaproteobacteria","c_Bacilli","c_Bacteroidia","c_Campylobacteria","c_Clostridia","c_Coriobacteriia","c_Cyanobacteriia","c_Desulfovibrionia","c_Fusobacteriia","c_Gammaproteobacteria","c_Incertae_Sedis","c_Methanobacteria","c_NA","c_Negativicutes","c_Synergistia","c_uncultured","c_Vampirivibrionia","c_Verrucomicrobiae","o_Acidaminococcales","o_Actinomycetales","o_Bacteroidales","o_Bifidobacteriales","o_Burkholderiales","o_Campylobacterales","o_Chloroplast","o_Christensenellales","o_Clostridia","o_Clostridia_UCG-014","o_Clostridia_vadinBB60_group","o_Clostridiales","o_Coriobacteriales","o_Corynebacteriales","o_Desulfovibrionales","o_DTU014","o_Enterobacterales","o_Erysipelotrichales","o_Eubacteriales","o_Flavobacteriales","o_Fusobacteriales","o_Gastranaerophilales","o_Izemoplasmatales","o_Lachnospirales","o_Lactobacillales","o_Methanobacteriales","o_Micrococcales","o_Monoglobales","o_NA","o_Oscillospirales","o_Paenibacillales","o_Pasteurellales","o_Peptococcales","o_Peptostreptococcales-Tissierellales","o_Pseudomonadales","o_RF39","o_Rhizobiales","o_Rhodospirillales","o_Staphylococcales","o_Synergistales","o_uncultured","o_Veillonellales-Selenomonadales","o_Verrucomicrobiales","f_[Clostridium]_methylpentosum_group","f_[Eubacterium]_coprostanoligenes_group","f_Acidaminococcaceae","f_Actinomycetaceae","f_Akkermansiaceae","f_Anaerofustaceae","f_Anaerovoracaceae","f_Atopobiaceae","f_Bacteroidaceae","f_Barnesiellaceae","f_Beijerinckiaceae","f_Bifidobacteriaceae","f_Butyricicoccaceae","f_Campylobacteraceae","f_Carnobacteriaceae","f_Chloroplast","f_Christensenellaceae","f_Clostridia_UCG-014","f_Clostridia_vadinBB60_group","f_Clostridiaceae","f_Coriobacteriaceae","f_Coriobacteriales_Incertae_Sedis","f_Corynebacteriaceae","f_Defluviitaleaceae","f_Desulfovibrionaceae","f_DTU014","f_Eggerthellaceae","f_Enterobacteriaceae","f_Enterococcaceae","f_Erysipelatoclostridiaceae","f_Erysipelotrichaceae","f_Eubacteriaceae","f_Flavobacteriaceae","f_Fusobacteriaceae","f_Gastranaerophilales","f_Gemellaceae","f_Hungateiclostridiaceae","f_Izemoplasmatales","f_Lachnospiraceae","f_Lactobacillaceae","f_Leuconostocaceae","f_Marinifilaceae","f_Methanobacteriaceae","f_Micrococcaceae","f_Monoglobaceae","f_Muribaculaceae","f_NA","f_Neisseriaceae","f_Oscillospiraceae","f_Oscillospirales","f_Oxalobacteraceae","f_Paenibacillaceae","f_Pasteurellaceae","f_Peptococcaceae","f_Peptostreptococcaceae","f_Peptostreptococcales-Tissierellales","f_Porphyromonadaceae","f_Prevotellaceae","f_Pseudomonadaceae","f_RF39","f_Rikenellaceae","f_Ruminococcaceae","f_Selenomonadaceae","f_Streptococcaceae","f_Sutterellaceae","f_Synergistaceae","f_Tannerellaceae","f_UCG-010","f_uncultured","f_Veillonellaceae","g_[Clostridium]_innocuum_group","g_[Clostridium]_methylpentosum_group","g_[Eubacterium]_brachy_group","g_[Eubacterium]_coprostanoligenes_group","g_[Eubacterium]_eligens_group","g_[Eubacterium]_fissicatena_group","g_[Eubacterium]_hallii_group","g_[Eubacterium]_nodatum_group","g_[Eubacterium]_ruminantium_group","g_[Eubacterium]_siraeum_group","g_[Eubacterium]_ventriosum_group","g_[Eubacterium]_xylanophilum_group","g_[Ruminococcus]_gauvreauii_group","g_[Ruminococcus]_gnavus_group","g_[Ruminococcus]_torques_group","g_Acidaminococcus","g_Actinomyces","g_Adlercreutzia","g_Agathobacter","g_Akkermansia","g_Alistipes","g_Alloprevotella","g_Anaerococcus","g_Anaerofilum","g_Anaerofustis","g_Anaerosporobacter","g_Anaerostipes","g_Anaerotruncus","g_Angelakisella","g_Atopobium","g_Bacteroides","g_Barnesiella","g_Bifidobacterium","g_Bilophila","g_Blautia","g_Butyricicoccus","g_Butyricimonas","g_Butyrivibrio","g_CAG-352","g_CAG-56","g_Campylobacter","g_Candidatus_Soleaferrea","g_Candidatus_Stoquefichus","g_Catenibacillus","g_Catenibacterium","g_CHKCI002","g_Chloroplast","g_Christensenella","g_Christensenellaceae_R-7_group","g_Cloacibacillus","g_Clostridia_UCG-014","g_Clostridia_vadinBB60_group","g_Clostridium_sensu_stricto_1","g_Colidextribacter","g_Collinsella","g_Coprobacillus","g_Coprobacter","g_Coprococcus","g_Corynebacterium","g_Defluviitaleaceae_UCG-011","g_Desulfovibrio","g_Dialister","g_Dielma","g_Dorea","g_DTU014","g_DTU089","g_Eggerthella","g_Eisenbergiella","g_Enterococcus","g_Enterorhabdus","g_Erysipelatoclostridium","g_Erysipelotrichaceae_UCG-003","g_Escherichia-Shigella","g_Eubacterium","g_Faecalibacterium","g_Faecalitalea","g_Family_XIII_AD3011_group","g_Family_XIII_UCG-001","g_Flavonifractor","g_Fournierella","g_Frisingicoccus","g_Fusicatenibacter","g_Fusobacterium","g_Gastranaerophilales","g_GCA-900066575","g_GCA-900066755","g_Gemella","g_Gordonibacter","g_Granulicatella","g_Haemophilus","g_Holdemanella","g_Holdemania","g_Howardella","g_Hungatella","g_Hydrogenoanaerobacterium","g_Incertae_Sedis","g_Intestinibacter","g_Intestinimonas","g_Izemoplasmatales","g_Lachnoclostridium","g_Lachnospira","g_Lachnospiraceae_FCS020_group","g_Lachnospiraceae_ND3007_group","g_Lachnospiraceae_NK4A136_group","g_Lachnospiraceae_NK4B4_group","g_Lachnospiraceae_UCG-001","g_Lachnospiraceae_UCG-004","g_Lachnospiraceae_UCG-008","g_Lachnospiraceae_UCG-009","g_Lachnospiraceae_UCG-010","g_Lactobacillus","g_Lactococcus","g_Lactonifactor","g_Leuconostoc","g_Marvinbryantia","g_Megasphaera","g_Merdibacter","g_Methanobrevibacter","g_Methylobacterium-Methylorubrum","g_Mogibacterium","g_Monoglobus","g_Moryella","g_Muribaculaceae","g_NA","g_Negativibacillus","g_Neisseria","g_NK4A214_group","g_Odoribacter","g_Oscillibacter","g_Oscillospira","g_Paenibacillus","g_Paludicola","g_Parabacteroides","g_Paraprevotella","g_Parasutterella","g_Parvimonas","g_Peptococcus","g_Peptoniphilus","g_Peptostreptococcus","g_Phascolarctobacterium","g_Phocea","g_Porphyromonas","g_Prevotella","g_Prevotellaceae_UCG-001","g_Pseudomonas","g_RF39","g_Romboutsia","g_Roseburia","g_Rothia","g_Ruminiclostridium","g_Ruminococcus","g_S5-A14a","g_Sellimonas","g_Senegalimassilia","g_Shuttleworthia","g_Slackia","g_Solobacterium","g_Streptococcus","g_Subdoligranulum","g_Sutterella","g_Terrisporobacter","g_Turicibacter","g_Tyzzerella","g_UBA1819","g_UC5-1-2E3","g_UCG-002","g_UCG-003","g_UCG-005","g_UCG-008","g_UCG-009","g_UCG-010","g_uncultured","g_Veillonella","s_[Clostridium]_leptum","s_[Clostridium]_scindens","s_[Clostridium]_spiroforme","s_[Eubacterium]_siraeum","s_Absiella_dolichum","s_Actinomyces_graevenitzii","s_Alistipes_indistinctus","s_Alistipes_inops","s_Alistipes_obesi","s_Alistipes_shahii","s_Alistipes_sp.","s_Anaerofustis_stercorihominis","s_Anaerotruncus_sp.","s_bacterium_YE57","s_Bacteroides_eggerthii","s_Bacteroides_intestinalis","s_Bacteroides_massiliensis","s_Bacteroides_ovatus","s_Bacteroides_plebeius","s_Bacteroides_salyersiae","s_Bacteroides_sp.","s_Bacteroides_stercoris","s_Bacteroides_vulgatus","s_Blautia_hydrogenotrophica","s_Blautia_sp.","s_Christensenella_massiliensis","s_Christensenella_minuta","s_Christensenella_sp.","s_Cloacibacillus_evryensis","s_Clostridiales_bacterium","s_Clostridium_perfringens","s_Clostridium_sp.","s_Colidextribacter_massiliensis","s_Collinsella_stercoris","s_Coprococcus_eutactus","s_Corynebacterium_durum","s_Desulfovibrio_desulfuricans","s_Eubacterium_ramulus","s_Faecalitalea_cylindroides","s_gut_metagenome","s_human_gut","s_Lachnospiraceae_bacterium","s_Lactococcus_lactis","s_Massiliomicrobiota_timonensis","s_Massiliprevotella_massiliensis","s_metagenome","s_Mogibacterium_diversum","s_Mordavella_sp.","s_NA","s_Parabacteroides_distasonis","s_Parabacteroides_johnsonii","s_Parabacteroides_merdae","s_Porphyromonas_asaccharolytica","s_Ruminococcaceae_bacterium","s_Ruminococcus_bicirculans","s_Ruminococcus_lactaris","s_Ruminococcus_sp.","s_Schaalia_odontolytica","s_Solobacterium_moorei","s_Streptococcus_mutans","s_Streptococcus_salivarius","s_Tyzzerella_sp.","s_uncultured_Alistipes","s_uncultured_Anaerovorax","s_uncultured_bacterium","s_uncultured_Bacteroidales","s_uncultured_Blautia","s_uncultured_Clostridia","s_uncultured_Clostridiaceae","s_uncultured_Clostridiales","s_uncultured_Clostridium","s_uncultured_Dorea","s_uncultured_Eubacteriaceae","s_uncultured_Eubacterium","s_uncultured_Firmicutes","s_uncultured_Lachnospiraceae","s_uncultured_marine","s_uncultured_organism","s_uncultured_Peptococcaceae","s_uncultured_prokaryote","s_uncultured_Roseburia","s_uncultured_rumen","s_uncultured_Ruminococcaceae","s_uncultured_Ruminococcus","s_uncultured_Thermoanaerobacterales","s_uncultured_Veillonellaceae","s_unidentified","o_Peptostreptococcales.Tissierellales","o_Clostridia_UCG.014","o_Veillonellales.Selenomonadales","f_Peptostreptococcales.Tissierellales","f_.Eubacterium._coprostanoligenes_group","f_.Clostridium._methylpentosum_group","f_Clostridia_UCG.014","f_UCG.010","g_S5.A14a","g_Lachnospiraceae_UCG.010","g_.Eubacterium._xylanophilum_group","g_Methylobacterium.Methylorubrum","g_.Eubacterium._nodatum_group","g_.Eubacterium._ventriosum_group","g_Lachnospiraceae_UCG.001","g_CAG.56","g_Erysipelotrichaceae_UCG.003","g_GCA.900066755","g_Prevotellaceae_UCG.001","g_UCG.002","g_GCA.900066575","g_UCG.008","g_.Eubacterium._coprostanoligenes_group","g_.Ruminococcus._gnavus_group","g_.Eubacterium._fissicatena_group","g_Family_XIII_UCG.001","g_.Eubacterium._brachy_group","g_.Ruminococcus._gauvreauii_group","g_CAG.352","g_.Ruminococcus._torques_group","g_.Clostridium._methylpentosum_group","g_.Eubacterium._hallii_group","g_UCG.009","g_UCG.005","g_.Eubacterium._eligens_group","g_Clostridia_UCG.014","g_.Eubacterium._ruminantium_group","g_UC5.1.2E3","g_Escherichia.Shigella","g_Lachnospiraceae_UCG.004","g_Lachnospiraceae_UCG.008","g_.Eubacterium._siraeum_group","g_UCG.003","g_UCG.010","g_Christensenellaceae_R.7_group","g_.Clostridium._innocuum_group","g_Lachnospiraceae_UCG.009","g_Defluviitaleaceae_UCG.011","s_.Clostridium._spiroforme","s_.Clostridium._scindens","s_.Clostridium._leptum","s_.Eubacterium._siraeum"],[false,false,false,true,false,false,true,true,false,false,false,false,false,true,false,false,true,false,true,true,true,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,true,false,true,false,true,false,true,true,false,false,false,false,true,false,true,false,false,false,false,false,false,true,false,false,false,true,false,false,false,false,true,false,false,false,false,true,false,false,false,false,false,false,true,false,true,false,false,true,true,false,true,false,false,true,false,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,true,false,false,false,false,false,true,false,true,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,true,true,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,true,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,true,false,false,true,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[false,true,false,true,false,false,true,true,true,false,false,false,false,true,false,false,true,false,true,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,true,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,true,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,true,true,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,true,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,true,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[false,false,true,false,true,false,false,true,false,false,false,false,false,false,false,true,false,true,false,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,null,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,null,false,false,false,false,true,false,false,null,false,null,null,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,null,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,null,true,false,false,false,false,false,false,false,false,false,false,null,false,false,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,null,null,false,false,false,false,false,false,false,false,null,false,null,false,false,false,false,false,false,false,false,null,false,false,false,false,false,false,false,false,false,false,false,null,null,false,false,false,false,null,false,false,true,true,true,false,null,null,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,null,null,null,null,null,false,false,false,false,false,false,false,false,null,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,true,true,false,false,true,false,null,false,false,false,false,false,false,false,null,false,false,false,false,false,false,false,false,false,false,false,false,null,null,null,null,null,null,null,false,false,null,null,null,null,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,true,false,false,false,false,true,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>TaxaName<\/th>\n      <th>ANCOMBC<\/th>\n      <th>ALDEx2<\/th>\n      <th>MaAsLin2<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}],"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="section level2">
<h2 id="step-7--adding-to-c3na-object">Step 7. Adding to C3NA Object<a class="anchor" aria-label="anchor" href="#step-7--adding-to-c3na-object"></a>
</h2>
<p>Now the DASummaryTable can be easily added to the C3NA object after the comparePhenotypes() function. The Shiny program will use the TaxaName (first column), and the remaining columns as selectable differential abundance methods in TRUE/FALSE format. Please make sure you only has the TaxaName column followed by a number of DA methods (with unique column names)</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## From Tutorial with Demo Step 4:</span>
<span class="va">CancerVsNormal_C3NA</span> <span class="op">=</span> <span class="fu"><a href="../reference/comparePhenotypes.html">comparePhenotypes</a></span><span class="op">(</span>C3NAObj_Comparison <span class="op">=</span>  <span class="va">Cancer_C3NA</span>, 
                                        C3NAObj_Reference <span class="op">=</span> <span class="va">Normal_C3NA</span>,
                                        corCutoff <span class="op">=</span> <span class="fl">0.2</span>, fdr <span class="op">=</span> <span class="fl">0.05</span>, unusefulTaxa <span class="op">=</span> <span class="cn">NA</span>, 
                                        nBootstrap <span class="op">=</span> <span class="fl">100</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="va">CancerVsNormal_C3NA</span><span class="op">$</span><span class="va">DA</span> <span class="op">=</span> <span class="va">DASummaryTable</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Kuncheng Song, Yi-Hui Zhou.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
